libraries/dotnet/EnergyOriginAuthorization/EnergyOriginAuthorization/EnergyOriginAuthorization/AuthorizeAttribute.cs

```diff
@@ -7,37 +6,25 @@ namespace EnergyOriginAuthorization
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true)]
     public class AuthorizeAttribute : Attribute, IAuthorizationFilter
     {
-        private IEnumerable<string> requiredScopes;
+        private readonly IEnumerable<string> requiredScopes;
         public AuthorizationContext? Context { get; private set; }
 
-        public AuthorizeAttribute()
-        {
-            requiredScopes = new List<string>();
-        }
+        public AuthorizeAttribute() => requiredScopes = new List<string>();
 
-        public AuthorizeAttribute(string requiredScope)
-        {
-            requiredScopes = new List<string>() { requiredScope };
-        }
+        public AuthorizeAttribute(string requiredScope) => requiredScopes = new List<string>() { requiredScope };
 
-        public AuthorizeAttribute(IEnumerable<string> requiredScopes)
-        {
-            this.requiredScopes = requiredScopes;
-        }
+        public AuthorizeAttribute(IEnumerable<string> requiredScopes) => this.requiredScopes = requiredScopes;
 
         public void OnAuthorization(AuthorizationFilterContext context)
         {
             var token = context.HttpContext.Request.Headers["Authorization"].FirstOrDefault()?.Split(" ").Last() ?? "";
 
             if (!ValidateToken(token, requiredScopes))
             {
-                context.Result = new JsonResult(new { message = "Unauthorized" }) { StatusCode = StatusCodes.Status401Unauthorized };
+                context.Result = new UnauthorizedObjectResult(new { message = "Unauthorized" });
```

Comment:
This is fine. UnauthorizedObjectResult will produce a response with status code 401.
The inheritance chain for JsonResult is: Object ->ActionResult -> JsonResult. 
The inheritance chain for UnauthorizedObjectResult is: Object -> ActionResult -> ObjectResult -> UnauthorizedObjectResult